### PR TITLE
Linting and Formatting Upgrade

### LINKED ISSUE #18876: Enable linting for the docs config and custom theme / components
Apparently, `eslint` ignores folders starting with `.` by default. This in turn means that the linter never checked any of the custom components in our Docs. We should make sure eslint also catches the `/docs/.vitepress/config.js` file, and everything inside `/docs/.vitepress/theme` and `/docs/.vitepress/components`

### PR DESCRIPTION
**Pull request by connorwinston**
See https://github.com/directus/directus/pull/18892#issuecomment-1607633804 for updated description.

---

- Enabled linting of files in `/docs/.vitepress` and `/docs/.typedocs`
- Enabled linting of `.md` mono-repo wide (Added `eslint-plugin-md`)
- Fixed formatting of  all`.md` files mono-repo wide
- Added `prettier-ignore` comments to invalid docs code blocks. 
  - You can now run `pnpm format` mono-repo wide without fear of messing up docs comments.
- Manually checked all docs pages to ensure no formatting got messed up in this upgrade. @phazonoverload 😀

Closes #18876

### CODE REVIEW

**Comment by paescuj** [comment]
Thanks @ConnorSimply, this is great ❤️ I'm going to adjust a few things in this PR and will mark it as "Ready for review" again afterwards.

**Comment by rijkvanzanten** [comment]
> Thanks @ConnorSimply, this is great ❤️ I'm going to adjust a few things in this PR and will mark it as "Ready for review" again afterwards.

What are you gonna change? Lets just put the needed changes in a review request, otherwise nobody knows what we're waiting for 😅

**Comment by ConnorSimply** [comment]
@paescuj Is this ready to be reviewed and merged?

File: docs/.vitepress/components/ArticleFeedback.vue [hunk 1]
```diff
@@ -81,7 +83,7 @@ async function handleSubmission(rating?: number) {
 		}
 	} catch (err) {
 		error.value = err;
-		console.error(err);
+		// console.error(err);
```
Comment (paescuj) [inline docs/.vitepress/components/ArticleFeedback.vue:86]: I think we can remove this altogether
```suggestion
```

**Comment by paescuj** [comment]
Sooo finally my owed explanation 🎉 

Updating the ESLint/Prettier setup is something I have been meaning to do for a while and I wanted to continue on Connor's initiative (thanks again ❤️) to make it more usable for the docs.
The final setup now consists of the following:

- No longer run Prettier via ESLint. Nowadays, this is the recommended way.
  - https://prettier.io/docs/en/integrating-with-linters.html#notes sums up the motivation behind it pretty well.
  - Makes the setup more flexible in general and allows us to easily run Prettier over the whole codebase be it via the CLI / Editor / CI.

- Enforce formatting via GitHub workflow.

- Disable formatting of code blocks in the docs.
  - This has the potential to break too much things, like `[!code ..]` comments.
  - Safer to turn it off completely instead of for individual blocks. Adding `prettier-ignore` comments to prevent formatting can be easily forgotten and at the same time it's hard to spot when things have been messed up.
 
- Clean-up as well as fixing some issues in the ESLint configuration.
  - For example some rules weren't applied to Vue files before (like `prefer-const`).
  - Clearer separation of the rules.
  - Removed `jest` plugin:  Wasn't configured correctly anyway and will be superfluous in the future.

- Add linting & formatting configuration for VS Code.
  - We can say pretty confidently that VS Code is the most used editor for devs working on our codebase, so I think it makes sense to add a configuration for it.
  - Since we require linting & formatting anyway, it makes sense to enable this by default for the editor.

- Enable local caching for both ESLint and Prettier to speed up things.

- Make editorconfig more useful.
  - It doesn't make sense to enforce tabs for _all_ files.
  - Now files can also be checked, for example via `editorconfig-checker`. Helped me to find issues like https://github.com/directus/directus/pull/18890/commits/337227d29162f64e55beeb7cf6d4838776bebbf8.

- Clean-up of ignore files.
  - Still enabling linting of files in `/docs/.vitepress` and `/docs/.typedocs` and additionally enable formatting for `en-US.yaml` file. 

---

Linting and formatting of code blocks in Markdown files is interesting but we cannot have it enabled by default.
It interferes too much, can mess things up and gives too many false negatives.
On the other side it proved to be helpful in finding quite some errors (syntax issues, wrong code block types, ...) and to improve formatting. For this reason I've left the possibility to check it manually:
```
pnpm --filter docs check-code-blocks
```

---

All lint & format issues have been fixed and double-checked.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**